dev/Breadcrumb/Breadcrumb.cpp

```diff
@@ -0,0 +1,561 @@
+    // the arrange method, so we retrieve the list from it
+    if (const auto& breadcrumbItemRepeater = m_breadcrumbItemRepeater.get())
+    {
+        if (const auto& breadcrumbLayout = breadcrumbItemRepeater.Layout().try_as<BreadcrumbLayout>())
+        {
+            return breadcrumbLayout->HiddenElements();
+        }
+    }
+
+    // By default just return an empty list
+    return winrt::make<Vector<winrt::IInspectable>>();
+}
+
+// When focus comes from outside the Breadcrumb control we will put focus on the selected item.
+void Breadcrumb::OnGettingFocus(const winrt::IInspectable&, const winrt::GettingFocusEventArgs& args)
+{
+    if (auto const& repeater = m_breadcrumbItemRepeater.get())
+    {
+        auto const& inputDevice = args.InputDevice();
+        if (inputDevice == winrt::FocusInputDeviceKind::Keyboard)
+        {
+            // If focus is coming from outside the repeater, put focus on the selected item.
+            auto const& oldFocusedElement = args.OldFocusedElement();
+            if (!oldFocusedElement || repeater != winrt::VisualTreeHelper::GetParent(oldFocusedElement))
+            {
+                // If the last focused element is now hidden, then focus the ellipsis button
+                if (auto const& repeaterLayout = repeater.Layout())
+                {
+                    if (auto const& breadcrumbLayout = repeaterLayout.try_as<BreadcrumbLayout>())
+                    {
+                        if (breadcrumbLayout->EllipsisIsRendered() &&
+                            m_focusedIndex < (int)breadcrumbLayout->FirstRenderedItemIndexAfterEllipsis())
+                        {
+                            FocusElement(0);
+                        }
+                    }
+                }
+
+                if (auto const& selectedItem = repeater.TryGetElement(m_focusedIndex))
+                {
+                    if (auto const& argsAsIGettingFocusEventArgs2 = args.try_as<winrt::IGettingFocusEventArgs2>())
+                    {
+                        if (args.TrySetNewFocusedElement(selectedItem))
+                        {
+                            args.Handled(true);
+                        }
+                    }
+                }
+            }
+
+            // Focus was already in the repeater: in On RS3+ Selection follows focus unless control is held down.
+            else if (SharedHelpers::IsRS3OrHigher() &&
+                (winrt::Window::Current().CoreWindow().GetKeyState(winrt::VirtualKey::Control) &
+                    winrt::CoreVirtualKeyStates::Down) != winrt::CoreVirtualKeyStates::Down)
+            {
+                if (auto const& newFocusedElementAsUIE = args.NewFocusedElement().as<winrt::UIElement>())
+                {
+                    FocusElement(repeater.GetElementIndex(newFocusedElementAsUIE));
+                    args.Handled(true);
+                }
+            }
+        }
+    }
+}
+
+void Breadcrumb::FocusElement(int index)
+{
+    if (index >= 0)
```

Comment:
When is index < 0?

Reply:
this is the line that may try to focus the -1 element

FocusElement(itemsRepeater.GetElementIndex(newFocusedElementAsUIE));